python/shotgun_model/shotgunmodel.py

```diff
@@ -948,6 +958,12 @@ def __rebuild_whole_tree_from_sg_data(self, data):
         root = self.invisibleRootItem()
         self.__populate_complete_tree_r(data, root, self.__hierarchy, {})
 
+        # Turn our signals back on as we have effectively reset the model at
+        # this point and finally call the QStandardItemModel's endResetModel
+        # method.
+        self.blockSignals(False)
```

Comment:
Is there a reason this isn't _before_ the re-population of the tree from the data?  It it were then this could all go into `__reset_all_data()` which has now been renamed to the public `clear()` in the latest version of the framework.  The behaviour of rows being added to the model would then be consistent regardless of which code path was taken here: https://github.com/tonybarbieri/tk-framework-shotgunutils/blob/qstandarditemmodel_signals/python/shotgun_model/shotgunmodel.py#L690 - what do you think?